java/src/main/kotlin/io/spine/validation/java/MessageValidationCode.kt

```diff
@@ -145,46 +159,93 @@ internal class MessageValidationCode(
     /**
      * Declares `validate()` method in this [MessagePsiClass].
      *
-     * This method does the following to address the formatting issues:
+     * This is a `public` implementation of [ValidatableMessage.validate] with
+     * [Override] annotation. The actual constraints are contained in its
+     * private overloading, that accepts the field path.
      *
-     * 1. [constraints] are joined to [String] using [joinToString] instead of `joinByLine()`
-     * because the contained code blocks already have new lines when converted to string.
-     * 2. For the same reason, we trim the result because a trailing empty line is not needed.
-     * 3. When creating a PSI method from text, we must use [trimMargin] with '|' symbols
-     * (the default separator) to explicitly specify the minimal common indent. The code blocks
-     * in the constraints add some whitespace characters, which break [trimIndent] and distort
-     * the resulting formatting, often leading to non-compilable Java code.
+     *  @see declarePrivateValidateMethod
      */
-    private fun MessagePsiClass.declareValidateMethod() {
-        val formattedConstraints = constraints.joinToString(separator = "").trim()
+    private fun MessagePsiClass.declarePublicValidateMethod() {
         val psiMethod = elementFactory.createMethodFromText(
             """
             |public java.util.Optional<io.spine.validate.ValidationError> validate() {
-            |    ${validateMethodBody(formattedConstraints)}
+            |    var noParent = io.spine.base.FieldPath.getDefaultInstance();
+            |    return validate(noParent);
             |}
-            """.trimMargin(), this
-        )
+            """.trimMargin(), this)
         psiMethod.annotate(Override::class.java)
         addLast(psiMethod)
     }
 
-    private fun validateMethodBody(formattedConstraints: String): String =
-        if (formattedConstraints.isEmpty()) {
-            "return java.util.Optional.empty();"
-        } else {
+    /**
+     * Declares `validate(FieldPath)` method in this [MessagePsiClass].
+     *
+     * All actual validations are rendered in this `private` overloading. It accepts
```

Comment:
I don't understand what this and the next paragraph mean. Something is missing. And "validations" is not a thing.